app/utils/url-helpers.js

```diff
@@ -0,0 +1,17 @@
+const { URL } = require('url')
+
+const parseUrlFromReq = (req) => {
+  const baseUrl = req.protocol + '://' + req.headers.host + '/'
+  return new URL(req.url, baseUrl)
+}
+
+const getQueryStringForReq = (req) => {
+  const requestUrl = parseUrlFromReq(req)
+  // remove leading `?`
+  return requestUrl.search.substring(1)
```

Comment:
Can we check for null before doing a substring ? As per interface null is possible

```
interface Url {
        ...
        search: string | null;
        ...
    }
```